learn/testing-ballerina-code/mocking.md

```diff
@@ -0,0 +1,418 @@
+---
+layout: ballerina-left-nav-pages
+title: Mocking
+description: Learn how to use Ballerina's built-in mocking API provided by the test to test modules
+ independent from other modules and external endpoints.
+keywords: ballerina, programming language, testing, mocking
+permalink: /learn/testing-ballerina-code/mocking/
+active: mocking
+redirect_from:
+  - /v1-2/learn/testing-ballerina-code/mocking
+  - /learn/testing-ballerina-code/mocking
+---
+
+# Mocking
+
+Mocking is useful to control the behavior of functions and objects to control the communication with other modules and external endpoints. A mock can be created by defining return values or replacing the entire object or function with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
+
+
+## Mocking Objects
+
+The test module provides capabilities to mock an object for unit testing. This allows you to control the behavior of object member functions and values of member fields via stubbing or replacing the entire object with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
```

Comment:
```suggestion
The test module provides capabilities to mock an object for unit testing. This allows you to control the behavior of the object member functions and values of member fields via stubbing or replacing the entire object with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
```
This sentence is repeated in the above "Mocking" section also. Shall we remove one of the occurrences?